transport_nantes/topicblog/tests_topic_blog_form.py

```diff
@@ -0,0 +1,192 @@
+        )
+        self.user_editor.is_staff = True
+        self.user_editor.save()
+        # Create a user that cant self publish editor 2
+        self.user_editor_two = User.objects.create_user(username='editor',
+                                                        password='editor')
+        self.user_editor_two.user_permissions.add(
+            edit_permission,
+            view_permission,
+            publish_permission,
+        )
+        self.user_editor_two.is_staff = True
+        self.user_editor_two.save()
+        # Create client for user
+        self.user_editor_client = Client()
+        self.user_editor_two_client = Client()
+        # Log user into the client
+        self.client.login(username='test-staff', password='test-staff')
+        self.user_editor_client.login(
+            username='test-editor', password='test-editor')
+        self.user_editor_two_client.login(
+            username='editor', password='editor')
+        # Take the session id cookie for log into selenium
+        self.cookie_admin = self.client.cookies['sessionid'].value
+        self.cookie_editor = self.user_editor_client.cookies['sessionid'].value
+        self.cookie_editor_two = (self.user_editor_two_client
+                                  .cookies['sessionid'].value)
+        """ Launch the browser the path of the browser will be
+         auto added with the webdriver_manager package and wait 10 sec
+        """
+        self.selenium = WebDriver(ChromeDriverManager().install())
+        self.selenium.implicitly_wait(10)
+
+    def tearDown(self):
+        # Close the browser
+        self.selenium.quit()
+
+    def fill_the_form_and_publish(self, slug, title):
+        """Fill the topicblog item form and publish
+        """
+        slug_input = self.selenium.find_element_by_name("slug")
+        slug_input.send_keys(slug)
+        title_input = self.selenium.find_element_by_name("title")
+        title_input.send_keys(title)
+        select = Select(self.selenium.find_element_by_name("template"))
+        select.select_by_value("topicblog/content.html")
+        self.selenium.find_element_by_link_text("Contenu (1)").click()
+        body_text_1_md_input = self.selenium.find_element_by_id(
+            "id_body_text_1_md")
+        body_text_1_md_input.send_keys("body 1")
+        body_text_2_md_input = self.selenium.find_element_by_id(
+            "id_body_text_2_md")
+        body_text_2_md_input.send_keys("body 2")
+        self.selenium.find_element_by_link_text("Contenu (2)").click()
+        body_text_3_md_input = self.selenium.find_element_by_id(
+            "id_body_text_3_md")
+        body_text_3_md_input.send_keys("body 3")
+        self.selenium.find_element_by_name("sauvegarder").click()
+        self.selenium.find_element_by_xpath(
+            "//input[@value='Publier']").click()
+
+    def test_user_permited_create_auto_publish(self):
+        self.selenium.get('%s%s' % (self.live_server_url, '/tb/admin/t/new'))
```

Comment:
Use a reverse() function, as the path is subject to changes, it's safer to use the name given to the url.